packages/upload-api/src/types/delegations.ts

```diff
@@ -34,4 +44,32 @@ export interface DelegationsStorage<
   ) => Promise<
     Ucanto.Result<Ucanto.Delegation<Ucanto.Tuple<Cap>>[], Ucanto.Failure>
   >
+
+  /**
+   * Given a list of delegation CIDs, return a Ucanto Result with a map from
+   * some or all of the CIDs to a list of "revocation context CIDs" for a
+   * given CID - that is, a list of delegation CIDs that should no longer
+   * be considered valid proof for the given CID.
+   */
+  getRevocations: (
+    delegationCIDs: Ucanto.Link[]
+  ) => Promise<
+    Ucanto.Result<RevocationsToMeta, Ucanto.Failure>
+  >
```

Comment:
I wonder if following would be a lot simpler 

```ts
interface RevocationStore {
  getAll(query: Revocation['delegation'][]): Promise<Result<Revocation[], Failure>>
  addAll(revocations: Revocation[]): Promise<Result<Unit, Failure>>
}


interface Revocation {
  delegation: UCANLink
  scope: UCANLink
  cause: UCANLink
}
```

Reply:
I like it! will get this updated and shipped